From 13f5312ea9bf9611c2d46ac310c46edc88d5ff22 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Revol?= <revol@free.fr>
Date: Sun, 17 Jan 2016 15:50:52 +0100
Subject: Xtranssock.c: Haiku has BSD UNIX socket but not SUN_LEN (yet?)

The test seems quite incorrect and results in SUN_LEN as an undefined symbol on Haiku.

diff --git a/Xtranssock.c b/Xtranssock.c
index b06579c..61cb106 100644
--- a/Xtranssock.c
+++ b/Xtranssock.c
@@ -994,7 +994,7 @@ TRANS(SocketUNIXCreateListener) (XtransConnInfo ciptr, const char *port,
     sockname.sun_len = strlen(sockname.sun_path);
 #endif
 
-#if defined(BSD44SOCKETS) || defined(SUN_LEN)
+#if defined(BSD44SOCKETS) && defined(SUN_LEN)
     namelen = SUN_LEN(&sockname);
 #else
     namelen = strlen(sockname.sun_path) + offsetof(struct sockaddr_un, sun_path);
@@ -1868,7 +1868,7 @@ TRANS(SocketUNIXConnect) (XtransConnInfo ciptr,
     sockname.sun_len = strlen (sockname.sun_path);
 #endif
 
-#if defined(BSD44SOCKETS) || defined(SUN_LEN)
+#if defined(BSD44SOCKETS) && defined(SUN_LEN)
     namelen = SUN_LEN (&sockname);
 #else
     namelen = strlen (sockname.sun_path) + offsetof(struct sockaddr_un, sun_path);
-- 
2.37.3

